x/nexus/keeper/msg_server.go

```diff
@@ -111,43 +113,104 @@ func (s msgServer) DeregisterChainMaintainer(c context.Context, req *types.Dereg
+		for _, chain := range s.GetChains(ctx) {
+			s.activateChain(ctx, chain)
 		}
+	} else {
+		for _, chainStr := range req.Chains {
+			chain, ok := s.GetChain(ctx, chainStr)
+			if !ok {
+				return nil, fmt.Errorf("%s is not a registered chain", chainStr)
+			}
+			s.activateChain(ctx, chain)
 
-		if s.IsChainActivated(ctx, chain) {
-			continue
 		}
+	}
+	return &types.ActivateChainResponse{}, nil
+}
 
-		sumConsensusPower := sdk.ZeroInt()
-		maintainers := s.GetChainMaintainers(ctx, chain)
-
-		for _, maintainer := range maintainers {
-			validator := s.staking.Validator(ctx, maintainer)
+// DeactivateChain handles deactivate chains in case of emergencies
+func (s msgServer) DeactivateChain(c context.Context, req *types.DeactivateChainRequest) (*types.DeactivateChainResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
 
-			if validator == nil || !validator.IsBonded() || validator.IsJailed() {
-				continue
+	if strings.ToLower(req.Chains[0]) == "all" {
+		for _, chain := range s.GetChains(ctx) {
+			s.deactivateChain(ctx, chain)
+		}
+	} else {
+		for _, chainStr := range req.Chains {
+			chain, ok := s.GetChain(ctx, chainStr)
+			if !ok {
+				return nil, fmt.Errorf("%s is not a registered chain", chainStr)
 			}
 
-			sumConsensusPower = sumConsensusPower.AddRaw(validator.GetConsensusPower(s.staking.PowerReduction(ctx)))
+			s.deactivateChain(ctx, chain)
 		}
+	}
+	return &types.DeactivateChainResponse{}, nil
+}
+
+func (s msgServer) activateChain(ctx sdk.Context, chain exported.Chain) {
+	if s.IsChainActivated(ctx, chain) {
+		return
+	}
 
-		if utils.NewThreshold(sumConsensusPower.Int64(), s.staking.GetLastTotalPower(ctx).Int64()).GTE(s.GetParams(ctx).ChainActivationThreshold) {
-			s.Nexus.ActivateChain(ctx, chain)
-
-			s.Logger(ctx).Info(fmt.Sprintf("chain %s activated", chain.Name))
-			ctx.EventManager().EmitEvent(
-				sdk.NewEvent(
-					types.EventTypeChain,
-					sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
-					sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueActivated),
-					sdk.NewAttribute(types.AttributeKeyChain, chain.Name),
-				),
-			)
+	// no chain maintainer for cosmos chains
+	if s.axelarnet.IsCosmosChain(ctx, chain.Name) {
+		s.Nexus.ActivateChain(ctx, chain)
+		return
+	}
+
+	if !isMetActivationThreshold(ctx, s.Nexus, s.staking, chain) {
+		s.Logger(ctx).Info(fmt.Sprintf("activation threshold is not met for %s", chain.Name))
+		return
+	}
+
+	s.Nexus.ActivateChain(ctx, chain)
+
+	s.Logger(ctx).Info(fmt.Sprintf("chain %s activated", chain.Name))
+	ctx.EventManager().EmitEvent(
+		sdk.NewEvent(
+			types.EventTypeChain,
+			sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+			sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueActivated),
+			sdk.NewAttribute(types.AttributeKeyChain, chain.Name),
+		),
+	)
+}
+
+func (s msgServer) deactivateChain(ctx sdk.Context, chain exported.Chain) {
+	if !s.IsChainActivated(ctx, chain) {
+		return
+	}
```

Comment:
```suggestion
	if !s.IsChainActivated(ctx, chain) {
		s.Logger(ctx).Info(fmt.Sprintf("chain %s deactivated", chain.Name))
		return
	}
```

Also, no event need here either?

Reply:
should we emit if it already deactivated?